components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultOIDCClaimsCallbackHandler.java

```diff
@@ -703,15 +704,21 @@ private JWTClaimsSet setClaimsToJwtClaimSet(JWTClaimsSet.Builder jwtClaimsSetBui
             String claimValue = claimEntry.getValue().toString();
             String claimKey = claimEntry.getKey();
             if (isMultiValuedAttribute(claimValue)) {
-                JSONArray claimValues = new JSONArray();
-                String[] attributeValues = claimValue.split(Pattern.quote(ATTRIBUTE_SEPARATOR));
-                for (String attributeValue : attributeValues) {
-                    if (StringUtils.isNotBlank(attributeValue)) {
-                        claimValues.add(attributeValue);
+                if (claimEntry.getKey().equals(ADDRESS)) {
+                    if (jwtClaimsSet.getClaim(claimKey) == null) {
+                        jwtClaimsSetBuilder.claim(claimEntry.getKey(), claimEntry.getValue());
```

Comment:
Can't we take this segment out of the if-else block and invert the if condition and get rid of the else block?

Reply:
Seems we can't do that. when the claim key is address we parse jsonObject(claimEntry.getValue()) and for other multi valued calims pasrse json Array(claimValues)

Reply:
Seems we can't do that. when the claim key is address we parse jsonObject(claimEntry.getValue()) and for other multi valued calims pasrse json Array(claimValues)